sphinx/builders/html/__init__.py

```diff
@@ -1032,7 +1036,7 @@ def hasdoc(name: str) -> bool:
             logger.warning(__("error writing file %s: %s"), outfilename, err)
         if self.copysource and ctx.get('sourcename'):
             # copy the source file for the "show source" link
-            source_name = path.join(self.outdir, '_sources',
+            source_name = path.join(self.outdir, self.sourcedir,
```

Comment:
Please give a more descriptive name to this. I suppose this name will confuse developers (including me!). This name is very similar to `self.srcdir`. It would be better not to make a variable if you can't give a better name.

Reply:
I agree the name is confusing. I'll try to find another one. 

About having or not a variable for a single use, if you wanna be SOLID, yes you definitely need to be *open for extension and closed for modification*.

Reply:
I found some usage of `_sources` string in `themes/agogo/layout.html` and `themes/basic/sourcelink.html`. Please consider them also.

Reply:
I don't object to change it to a variable only if we can give a good name for it. I think extendability is based on good names. So we have to consider the better name to make it extendable.